maven/build.go

```diff
@@ -82,6 +84,24 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 		if err := os.Chmod(command, 0755); err != nil {
 			return libcnb.BuildResult{}, fmt.Errorf("unable to chmod %s\n%w", command, err)
 		}
+
+		file, err := ioutil.ReadFile(command)
+		if err != nil {
+			return libcnb.BuildResult{}, fmt.Errorf("unable to read mvnw file %s\n%w", command, err)
+		}
+
+		if bytes.ContainsAny(file, "\r\n") || bytes.ContainsAny(file, "\r") {
+
+			// the mvnw file can contain Windows CRLF line endings, e.g. from a 'git clone' on windows
+			// we call replaceLineEndings to replace these so that the unix container can execute the wrapper successfully
+			b := replaceLineEndings(file)
+
+			err = ioutil.WriteFile(command, []byte(b), 0755)
+			if err != nil {
+				return libcnb.BuildResult{}, fmt.Errorf("unable to process mvnw file format %s\n%w", command, err)
+			}
+		}
+
```

Comment:
I think it would be preferred to wrap this whole part inside the function. You could call it something like `CleanMvnWrapper`. It wouldn't really need to return anything, just replace the line endings if they exist and skip if they do not. Return `error` if there's an I/O problem or `nill`. Then the main function here, just calls `CleanMvnWrapper` and checks for an error. 

If you make it a public function, i.e. starts with a capital letter, like `CleanMvnWrapper` you can then directly test this function which will simplify your tests just a little too.

I'm also not 100% sure that you need to check for the presence of the line encodings you're going to replace. I think if you just read, attempt to replace, and write out you should be fine. Doing the replace on the bytes in memory isn't really going to be that time-consuming and if there's nothing to replace, it'll be a no-op.

Reply:
Made 3/3 changes